### PR TITLE
Fixed issues/improvements in unit testing framework

### PR DESCRIPTION
**Pull request by sajithaliyanage**
## Purpose
This PR is related to fixing some issues and improving the current flow of the unit testing framework. Following points are covered from this PR. 

- Observability improvement in Integration Studio when a test case failure. Now it shows the actual and expected test case assertion details in the unit test report. 
Related issue - https://github.com/wso2/devstudio-tooling-ei/issues/963
Screenshot 2020-05-06 at 11 02 50
Also it shows invoked URL and received status code of the test case failure service. 
Screenshot 2020-05-06 at 11 34 32

- Improved test case mediation without input payload for PUT and DELETE resources in an API.
Related issue - https://github.com/wso2/devstudio-tooling-ei/issues/962

- Add two new properties as `$statusCode`, `$httpVersion` to assert status code and HTTP version of the services.  
Related issue - https://github.com/wso2/devstudio-tooling-ei/issues/961
```xml
<assertions>
          <assertEquals>
                    <actual>$body</actual>
                    <expected><![CDATA[DELETE RESPONSE]]></expected>
                    <message>The response payload is not correct!</message>
          </assertEquals>
          <assertEquals>
                    <actual>$statusCode</actual>
                    <expected><![CDATA[200]]></expected>
                    <message>The HTTP Status code is not correct!</message>
           </assertEquals>
           <assertEquals>
                    <actual>$httpVersion</actual>
                    <expected><![CDATA[HTTP/1.1]]></expected>
                    <message>The HTTP version is not correct!</message>
           </assertEquals>
</assertions>
```

- Improve server logs of the unit testing agent. 
- Fixed issue in AssertNotNull assertion type. 
- Add error logs for mock service failures

### CODE REVIEW

File: modules/core/src/main/java/org/apache/synapse/unittest/TestingAgent.java [hunk 1]
```diff
@@ -99,7 +99,7 @@
                         isArtifactDeployed = true;
                         deploymentStats.put(key, TYPE_SEQUENCE);
                         testSuiteSummary.setDeploymentStatus(Constants.PASSED_KEY);
-                        log.info("Sequence artifact deployed successfully");
+                        log.info("Sequence main test artifact deployed successfully");
```
Comment (sdkottegoda) [inline modules/core/src/main/java/org/apache/synapse/unittest/TestingAgent.java:102]: What does main test mean? Why is it essential?

File: modules/core/src/main/java/org/apache/synapse/unittest/TestingAgent.java [hunk 2]
```diff
@@ -118,7 +118,7 @@
                         isArtifactDeployed = true;
                         deploymentStats.put(key, TYPE_PROXY);
                         testSuiteSummary.setDeploymentStatus(Constants.PASSED_KEY);
-                        log.info("Proxy artifact deployed successfully");
+                        log.info("Proxy main test artifact deployed successfully");
```
Comment (sdkottegoda) [inline modules/core/src/main/java/org/apache/synapse/unittest/TestingAgent.java:121]: Shouldn't this be something like 'Primary test Proxy artifact deployed successfully.' "I didn't understand what "Proxy main test artifact deployed successfully" at the beginning.

Comment (sajithaliyanage) [reply modules/core/src/main/java/org/apache/synapse/unittest/TestingAgent.java]: Done. Changed the logs as suggested.

File: modules/core/src/main/java/org/apache/synapse/unittest/TestingAgent.java [hunk 3]
```diff
@@ -152,7 +152,7 @@
 
             }
         } catch (Exception e) {
-            log.error("Artifact deployment failed", e);
+            log.error("Main test artifact " + mainTestArtifactType + " deployment failed", e);
```
Comment (sdkottegoda) [inline modules/core/src/main/java/org/apache/synapse/unittest/TestingAgent.java:155]: "Main" -> "Primary" here aswell?

File: modules/core/src/main/java/org/apache/synapse/unittest/TestingAgent.java [hunk 4]
```diff
@@ -178,14 +178,14 @@
...
-                    log.error("Artifact deployment failed", e);
+                    log.error("Supportive test artifact deployment failed", e);
                     exception = CommonUtils.stackTraceToString(e);
                     testSuiteSummary.setDeploymentStatus(Constants.FAILED_KEY);
                     testSuiteSummary.setDeploymentException(exception);
                 }
             } else {
                 log.error(synapseTestCase.getArtifacts().getSupportiveArtifact(x).getArtifactType()
-                        + " artifact deployment failed");
+                        + " main test artifact deployment failed");
```
Comment (sdkottegoda) [inline modules/core/src/main/java/org/apache/synapse/unittest/TestingAgent.java:188]: "Main" -> "Primary" here aswell?

Comment (sajithaliyanage) [reply modules/core/src/main/java/org/apache/synapse/unittest/TestingAgent.java]: Done

File: modules/core/src/main/java/org/apache/synapse/unittest/TestingAgent.java (hunk 3 above)
Comment (sajithaliyanage) [reply modules/core/src/main/java/org/apache/synapse/unittest/TestingAgent.java]: Done

**Pull request merged**